.github/workflows/test.yml

```diff
@@ -7,22 +7,140 @@ jobs:
+        run: (Get-Content pyproject.toml).replace('python = "^3.6"', 'python = "~${{ matrix.python-version }}"') | Set-Content pyproject.toml
+
+      - name: Force Flask version
+        run: poetry add Flask==${{ matrix.flask-version }}
+
+      - name: Overwrite Flask dependencies for legacy install
+        if: matrix.flask-version < '2.0' && matrix.flask-version != 'latest'
+        run: |
+          poetry add Jinja2=^2.0
+          poetry add Markupsafe==2.0.1
+          poetry add itsdangerous==2.0.1
+
+      - name: Overwrite Flask dependencies for legacy install (2)
+        if: matrix.flask-version < '2.1' && matrix.flask-version != 'latest'
+        run: poetry add werkzeug==2.0.3
+
       - name: Install requirements
         run: poetry install
+
+      - name: List installed package versions for manual inspection
+        run: python3 --version && poetry --version && poetry show
+
       - name: Run unit tests
         run: poetry run test
+
       - name: Run doctests
         run: poetry run doctest
+
+  check_pip_install:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.6', '3.7', '3.8', '3.9', '3.10']
+        flask-version: ['1.0', '1.1', '2.0', '2.1', 'latest']
+        os: [ubuntu-18.04, macos-latest]
+        exclude:
+          # starting from Flask 2.1.0, python 3.6 is no longer supported:
+          - flask-version: '2.1'
+            python-version: '3.6'
+
+          - flask-version: 'latest'
+            python-version: '3.6'
+
+          # old versions of Flask no longer working with python 3.10:
+          - flask-version: '1.0'
+            python-version: '3.10'
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Flask
+        run: |
+          if [[ "${{ matrix.flask-version }}" == "latest" ]]
+          then
+            pip3 install Flask
+          else
+            pip3 install Flask==${{ matrix.flask-version }}
+          fi
+
+      - name: Install current flask-selfdoc from GitHub
+        run: |
+          git_url=$(echo "${{ github.repositoryUrl }}@${{ github.head_ref }}" | sed "s/git:/git+https:/")
+          pip3 install ${git_url}
+
+      - name: Overwrite Flask dependencies for legacy install
+        if: matrix.flask-version < '2.0' && matrix.flask-version != 'latest'
+        run: |
+          pip3 install "Jinja2<3.0"
+          pip3 install MarkupSafe==2.0.1
+          pip3 install itsdangerous==2.0.1
+
+      - name: Overwrite Flask dependencies for legacy install (2)
+        if: matrix.flask-version < '2.1' && matrix.flask-version != 'latest'
+        run: pip3 install werkzeug==2.0.3
+
+      - name: List installed package versions for manual inspection
+        run: python3 --version && pip3 list
+
+      - name: Check that Flask version did not change
```

Comment:
control that `pip`'s dependency solver didn't need to update the already installed version of Flask